Automattic/jetpack
# Declare PHP version requirement in all Composer packages

## Proposed changes:

This makes for a better experience for third parties using the packages, as they can see on Packagist which PHP versions the package is supposed to support.

Also this is a good excuse to make a major version bump for all the packages following #34126 which removed testing with PHP 5.6 and allows use of 7.0+ features going forward.

And since we're making systematic use of declared PHP versions now, let's lint for that and make use of it to skip phpunit tests where appropriate.

### Other information:


## Jetpack product discussion


None really, although https://github.com/Automattic/jetpack/pull/34126#discussion_r1394352324 is somewhat related

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Is CI happy?
  * [x] Codesniffer tests skipped in PHP 7.0 with the new check for that. https://github.com/Automattic/jetpack/actions/runs/6907991966/job/18796345466?pr=34192#step:9:1639
  * [x] Codesniffer tests still ran in PHP 7.4 as expected. https://github.com/Automattic/jetpack/actions/runs/6907991966/job/18796346607?pr=34192#step:9:1781
* Try using `jetpack generate` to generate a package, it should include `.require.php` in composer.json.
* Try removing the `.require.php` from a package and see if `.github/files/lint-project-structure.sh` catches it with an appropriate message.
